tests/AbstractLoopTest.php

```diff
@@ -167,71 +195,76 @@ public function emptyRunShouldSimplyReturn()
     /** @test */
     public function runShouldReturnWhenNoMoreFds()
     {
-        $input = $this->createStream();
-
+        list ($input, $output) = $this->createSocketPair();
+        
         $loop = $this->loop;
         $this->loop->addReadStream($input, function ($stream) use ($loop) {
             $loop->removeStream($stream);
         });
-
-        $this->writeToStream($input, "foo\n");
-
+        
+        fwrite($output, "foo\n");
+        
         $this->assertRunFasterThan(0.015);
     }
 
     /** @test */
     public function stopShouldStopRunningLoop()
     {
-        $input = $this->createStream();
-
+        list ($input, $output) = $this->createSocketPair();
+        
         $loop = $this->loop;
         $this->loop->addReadStream($input, function ($stream) use ($loop) {
             $loop->stop();
         });
-
-        $this->writeToStream($input, "foo\n");
-
+        
+        fwrite($output, "foo\n");
+        
         $this->assertRunFasterThan(0.005);
     }
 
     public function testStopShouldPreventRunFromBlocking($timeLimit = 0.005)
     {
-        $this->loop->addTimer(
-            1,
-            function () {
-                $this->fail('Timer was executed.');
-            }
-        );
-
-        $this->loop->nextTick(
-            function () {
-                $this->loop->stop();
-            }
-        );
-
+        $this->loop->addTimer(1, function () {
+            $this->fail('Timer was executed.');
+        });
+        
+        $this->loop->nextTick(function () {
+            $this->loop->stop();
+        });
+        
```

Comment:
I will have to start from scratch to undo the formatting. I will look into it later today or maybe tomorrow.